samples/switchless/README.md

```diff
@@ -114,14 +132,21 @@ oeedger8r ../switchless.edl --untrusted
 
 ## About the host
 
-The host first defines a structure specifically for configuring switchless calls. In this case, we specify the
-first field `1` as the number of host worker threads for switchless OCALLs. In this example, 1) There is at most
-1 enclave thread all the time, and 2) The number of cores available to the host worker threads is unknown, and
-so we use 1 as explained above. The 2nd field specifies the number of enclave threads for switchless ECALLs.
-Since switchless ECALL is not yet implemented, we require the 2nd field to be `0`.
+The host first defines a structure specifically for configuring switchless calls.
+
+We conservatively assume that we are executing on a machine with 2 cores. The main thread will keep posting
+switchless ocalls from within the enclave. To service these ocalls, we will create one host worker thread by setting the first field (`max_host_workers`)
+of the structure to `1`. Since there is only one thread making synchronous switchless ocalls,
+there is no benefit to creating more that one worker thread. By restricting the number of caller and worker threads both to 1 on a two core machine, we ensure that the caller and worker threads can potentially run uninterrupted on
+each of the available cores, causing less contention, and leading to maximum speedup via switchless calling.
+
+Once all the switchless ocalls have been serviced, the main thread will return to the host and then keep posting switchless ecalls.
+Inorder to service these switchless ecalls, we will create one enclave worker thread by setting the second field (`max_enclave_workers`) of the structure to `1`. At this time the host worker thread that serviced the switchless ocalls will automatically go to sleep since there are no more ocalls for it to process. Just as in the case of ocalls, for ecalls, one enclave worker thread and one host caller thread would lead to maximum speedup via switchless calling in a two core machine. For a machine with more cores, the number of worker threads can be increased in accordance with the recommendations given earlier.
```

Comment:
Suggest splitting into multiple source lines to make diffs/reviewing easier

Reply:
Good point. Done.